src/main/java/com/example/WorldOfAnimals/controllers/AnimalImageController.java

```diff
@@ -0,0 +1,63 @@
+package com.example.WorldOfAnimals.controllers;
+
+import com.example.WorldOfAnimals.exceptions.AnimalImageNotFoundException;
+import com.example.WorldOfAnimals.services.AnimalImageService;
+import lombok.AllArgsConstructor;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+@RestController
+@RequestMapping("/animals/images")
+@AllArgsConstructor
+public class AnimalImageController {
+
+    private final AnimalImageService service;
+
+    @PostMapping
```

Comment:
@PostMapping(consumes = {MediaType.MULTIPART_FORM_DATA_VALUE})
    @ResponseStatus(HttpStatus.OK)
    public void uploadAnimalImage(
            @RequestPart @Valid @NotNull AnimalImageRequest request,
            @RequestPart @NotNull MultipartFile file
    ) {
        service.upload(request, file);
    }

Reply:
When you upload image, need send body with animal id

Reply:
done